### ISSUE TITLE
Timeout when sending a *IDN? command to a Siglent SDS2000 scope

### ISSUE THREAD

**Issue by hwstar** [python-ivi/python-vxi11#18]
I'm trying to get python-vxi11 working with all of my test equipment, and most of my instruments work great with it except for my Siglent SDS2000 scope. When sending the *IDN? command the ask command hangs, and then times out. I've included the wireshark file if it helps.

Tracing through the code, it seems to attempt to open a second connection to the instrument, and hangs up there.

This is what gets returned after a while:

 File "/home/srodgers/projects/python/PycharmProjects/dp832/instrument.py", line 60, in identify
    return self.s.ask("*IDN?")
  File "/usr/local/lib/python3.4/dist-packages/python_vxi11-0.8-py3.4.egg/vxi11/vxi11.py", line 717, in ask
  File "/usr/local/lib/python3.4/dist-packages/python_vxi11-0.8-py3.4.egg/vxi11/vxi11.py", line 702, in write
  File "/usr/local/lib/python3.4/dist-packages/python_vxi11-0.8-py3.4.egg/vxi11/vxi11.py", line 610, in write_raw
  File "/usr/local/lib/python3.4/dist-packages/python_vxi11-0.8-py3.4.egg/vxi11/vxi11.py", line 580, in open
  File "/usr/local/lib/python3.4/dist-packages/python_vxi11-0.8-py3.4.egg/vxi11/vxi11.py", line 493, in __init__
  File "/usr/local/lib/python3.4/dist-packages/python_vxi11-0.8-py3.4.egg/vxi11/rpc.py", line 522, in __init__
  File "/usr/local/lib/python3.4/dist-packages/python_vxi11-0.8-py3.4.egg/vxi11/rpc.py", line 257, in __init__
  File "/usr/local/lib/python3.4/dist-packages/python_vxi11-0.8-py3.4.egg/vxi11/rpc.py", line 261, in connect
TimeoutError: [Errno 110] Connection timed out

[wireshark.zip](https://github.com/python-ivi/python-vxi11/files/602582/wireshark.zip)

**Comment by alexforencich** [comment]
The Siglent scope does not appear to implement the VXI-11 protocol correctly.  It's supposed to implement an abort channel and report the port that this service is listening on in the create_link command.  It reports port 1, but then it seems the subsequent connection attempt to open the abort channel on port 1 times out as the device doesn't appear to be listening on that port.  

Now, there may be some reasonable way to work around this by not opening an abort channel.  Maybe just deferring the creation of the abort channel until abort is actually called (if it ever gets called).  I'll see about making that change.

**Comment by alexforencich** [comment]
Alright, try the code on my fork (https://github.com/alexforencich/python-vxi11/) and let me know what it does for you.

**Comment by hwstar** [comment]
Yes, that resolves the timeout issue. Thanks for the help! I've got a short list of other firmware issues with the Siglent scope which I'll be passing on to them.
